tools/ci-build/sdk-lockfiles/src/audit.rs

```diff
@@ -0,0 +1,430 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::AuditArgs;
+use anyhow::bail;
+use anyhow::{Context, Result};
+use cargo_lock::dependency::graph::{Graph, NodeIndex};
+use cargo_lock::{package::Package, Lockfile};
+use petgraph::visit::EdgeRef;
+use smithy_rs_tool_common::git::find_git_repository_root;
+use smithy_rs_tool_common::here;
+use std::collections::{BTreeSet, HashSet};
+use std::env;
+use std::iter;
+use std::path::PathBuf;
+
+// A list of AWS runtime crate must be in sync with
+// https://github.com/smithy-lang/smithy-rs/blob/0f9b9aba386ea3063912a0464ba6a1fd7c596018/buildSrc/src/main/kotlin/CrateSet.kt#L42-L53
+// plus `aws-inlineable`
+const AWS_SDK_RUNTIMES: &[&str] = &[
+    "aws-config",
+    "aws-credential-types",
+    "aws-endpoint",
+    "aws-http",
+    "aws-hyper",
+    "aws-inlineable",
+    "aws-runtime",
+    "aws-runtime-api",
+    "aws-sig-auth",
+    "aws-sigv4",
+    "aws-types",
+];
+
+// A list of server runtime crates must be in sync with
+// https://github.com/smithy-lang/smithy-rs/blob/0f9b9aba386ea3063912a0464ba6a1fd7c596018/buildSrc/src/main/kotlin/CrateSet.kt#L85-L87
+const SERVER_SPECIFIC_RUNTIMES: &[&str] = &[
+    "aws-smithy-http-server",
+    "aws-smithy-http-server-python",
+    "aws-smithy-http-typescript",
+];
+
+fn new_dependency_for_aws_sdk(crate_name: &str) -> bool {
+    AWS_SDK_RUNTIMES.contains(&crate_name)
+        || crate_name == "inlineable"
+        || (crate_name.starts_with("aws-smithy-")
+            && !SERVER_SPECIFIC_RUNTIMES.contains(&crate_name))
+}
+
+// Recursively traverses a chain of dependencies originating from a potential new dependency. Returns true as soon as
```

Comment:
Nit: The word dependency is a bit confusing as it indicates that we will look for the dependencies of the provided node, where as we are looking for crates that consume this dependency.

How about:

// This function checks if the given dependency is directly or indirectly consumed by any AWS SDK runtime crate.
// It recursively traverses the chain of consumers starting from the given Cargo dependency.
// Returns true as soon as it encounters a crate name that matches one of the AWS SDK runtime crates.

Reply:
I think the suggested comment is more for a comment for the `new_dependency` function. This comment `Recursively traverses...` is for the helper used by `new_dependency`.